src/app/api/v1/endpoints/create_pngs.py

```diff
@@ -29,7 +29,7 @@ def create_pngs(aws_filename: str = Form(...)):
         raise HTTPException(status_code=400, detail="Invalid request. The filename must end with '.pdf'.")
 
     # Get the filename from the path
-    filename = aws_filename.split(".")[0]
+    filename = aws_filename.replace(".pdf", "")
```

Comment:
```suggestion
    filename = aws_filename[:-4]
```
I would do it like this, since we just checked that the filename ends with ".pdf", we are safe to remove the last four characters. You're code has the disadvantage that we might delete ".pdf" also in the middle of the string, in the (admittedly unlikely) case that we have a filename where ".pdf" occurs not just as the file extension, but also somewhere in the middle of the filename.

Reply:
How about using a path specific function for this like here: https://stackoverflow.com/a/42288083/9543860

I think it would be "easier" to read and understand that we removed the suffix. What do you think? Otherwise, I am happy to merge the code with your suggestion.

Reply:
Yes, that's even better.
And then (for consistency) the check above can be written as `aws_filename.suffix == ".pdf"` instead of using the `endswith` function.